### PR TITLE
Adding `pulumi.IsSecret` and `pulumi.Unsecret` to the Go SDK

### PR DESCRIPTION
**Pull request by stack72**
Related: #5653

There are situations where we need to ensure we understand if an
Output contains secrets and for the ability to take an Output that
contains a secret and creates the unsecret version

```
secretString := ToSecret(String("foo"))

unSecretString := Unsecret(secretString)
```

### CODE REVIEW

File: sdk/go/pulumi/types.go [hunk 1]
```diff
@@ -419,11 +419,26 @@ func (o *OutputState) ApplyTWithContext(ctx context.Context, applier interface{}
...
+func Unsecret(input Output) Output {
+	return UnsecretWithContext(context.Background(), input)
+}
+
+// UnsecretWithContext will unwrap a secret output as a new output with a resolved value and no secretness
+func UnsecretWithContext(ctx context.Context, input Output) Output {
+	var x bool
+	x = false
+	o := toOutputWithContext(ctx, input, &x)
+	// set immediate secretness ahead of resolution/fufillment
```
Comment (lblackstone) [inline sdk/go/pulumi/types.go:437]:
```suggestion
	// set immediate secretness ahead of resolution/fulfillment
```

**Review by lblackstone — APPROVED** [review]

File: sdk/go/pulumi/types.go (hunk 1 above)
Comment (stack72) [reply sdk/go/pulumi/types.go:437]: Nice find - I'll merge this and fix that in my next PR

**Review by viveklak — APPROVED** [review]

**Comment by stack72** [comment]
Thanks @viveklak - I'll send a PR to fix these up as well

**Comment by viveklak** [comment]
> Thanks @viveklak - I'll send a PR to fix these up as well

No worries. Sorry for the late review. Forgot to hit submit.

**Comment by EvanBoyle** [comment]
Looks good, nice work! Sorry I was late on reviewing this.

**Pull request merged**